payroll/search.py

```diff
@@ -161,7 +170,7 @@ def _search(self, entity_type, *args):
                 self.facets = {entity_type: results.facets}
 
         # Retain ordering from Solr results when filtering the model objects.
-        sorted_results = OrderedDict([(self._id_from_result(r), r) for r in results])
+        sorted_results = collections.OrderedDict([(self._id_from_result(r), r) for r in results])
```

Comment:
@hancush I have a clear idea about how to improve the efficiency of this section. do you mind if i make a commit for that?

Reply:
i don't mind at all. go ahead.